RereleaseTimer/index.js

```diff
@@ -1,203 +1,100 @@
-                            context.log(`Sessions (1-30min): ${sessions}`);
-                            context.log(`Total Installs: ${installs}`);
-
-                            if (crashes <= rule.crashes && installs >= rule.installs && sessions >= rule.sessions) {
-                                context.log(`Re-releasing latest version...`);
-
-                                return GetDestination(token, owner, app, rule)
-                                .then(group => {
-                                    if(group) {
-                                        return GetRelease(token, owner, app, release.id)
-                                        .then(release => {
-                                            if(release) {
-                                                const patch = {
-                                                    destinations: [{ id: group.id, name: group.name }],
-                                                    mandatory_update: release.mandatory_update,
-                                                    release_notes: release.release_notes
-                                                };
-                                                PatchRelease(token, owner, app, release.id, patch)
-                                                .then((release) => {
-                                                    resolve()
-                                                })
-                                                .error(response => {
-                                                    context.error(response);
-                                                    reject(response);
-                                                });
-                                            }
-                                        });
-
-                                        resolve(true);
-                                    } else {
-                                        reject("Could not lookup destination for re-release.");
-                                    }
-                                });
-                            } else {
-                                context.log(`Nothing to perform.`);
-                                resolve(false);
-                            }
-                        });
-                    } else {
-                        context.log(`Latest release (${release.short_version}) has already been distributed to the destination.`);
-                        resolve(false);
-                    }
-                } else {
-                    context.log("No releases available in source.");
-                    resolve(false);
-                }
-            })
-            .error(response => {
-                reject(error);
-                context.error(response);
-            });
-        }));
+    for (rule of rules) {
+        ruleSet.push(getRulePromise(rule, context));
     }
 
     Promise.all(ruleSet)
```

Comment:
What if 2 rules are finished and 1 is failed, in this case we need to break down the whole process?

Reply:
Got it, I think we need to modify logic